### DISCUSSION TITLE
Assertion failed: 0

### DISCUSSION THREAD

**Discussion by mohammadmoth** [libuv/libuv#3444] [General]
hello all ,
I am getting this error while running in nodejs:

**Assertion failed: 0, file c:\ws\deps\uv\src\win\handle-inl.h, line 160**
 
Can you please explain to me what that means?

Thank you all

**Comment by bnoordhuis** [comment]
Basically that you hit an impossible condition, more specifically that a libuv handle was of an unrecognized type. Most likely there's some kind of memory corruption going on in your application, like freeing the handle's memory before libuv is done with it.

You appear to be using ws (which in turn uses libuv) so you should probably redirect your question to that project.

**Comment by mohammadmoth** [comment]
Thank you very much
Specifically we are using Native Abstractions for Node.js (Nan)
This means we can use libuv directly.
What you say is actually happening somewhere.

I would like to ask you something else. Sometimes nodejs crashed without saying anything!!
How can I trace the error in this case? What I actually brought up in the node .But there is no answer(https://github.com/nodejs/help/issues/3693)

Do you have any interesting idea or information about it?

We have a similar case: 
https://github.com/nodejs/help/issues/2169


Thank you .

**Reply by bnoordhuis** [reply]
I can't really help you with that but maybe you can post it over at https://github.com/nodejs/node/discussions - might attract more eyeballs there.

**Reply by mohammadmoth** [reply]
Thank you 
I will
